frontend/components/BookTable.tsx

```diff
@@ -0,0 +1,272 @@
+        </div>
+      ),
+    },
+    {
+      title: "Published Date",
+      dataIndex: "published_date",
+      sorter: (a: Book, b: Book) =>
+        new Date(a.published_date).getTime() -
+        new Date(b.published_date).getTime(),
+      filterDropdown: ({
+        setSelectedKeys,
+        selectedKeys,
+        confirm,
+        clearFilters,
+      }: {
+        setSelectedKeys: (selectedKeys: string[]) => void;
+        selectedKeys: string[];
+        confirm: () => void;
+        clearFilters: () => void;
+      }) => (
+        <div style={{ padding: 8 }}>
+          <RangePicker
+            value={selectedKeys[0] as any}
+            onChange={(dates) => setSelectedKeys(dates ? [dates as any] : [])}
+            style={{ marginBottom: 8, display: "block" }}
+          />
+          <Button
+            type="primary"
+            onClick={() => confirm()}
+            size="small"
+            style={{ width: 90, marginRight: 8 }}
+          >
+            Filter
+          </Button>
+          <Button
+            onClick={() => {
+              clearFilters();
+            }}
+            size="small"
+            style={{ width: 90 }}
+          >
+            Reset
+          </Button>
+        </div>
+      ),
+    },
+    {
+      title: "Author",
+      dataIndex: "author",
+      render: (author: any) => author?.name || "-",
+      filterDropdown: ({
+        setSelectedKeys,
+        selectedKeys,
+        confirm,
+        clearFilters,
+      }: {
+        setSelectedKeys: (selectedKeys: string[]) => void;
+        selectedKeys: string[];
+        confirm: () => void;
+        clearFilters: () => void;
+      }) => (
+        <div style={{ padding: 8 }}>
+          <Input
+            placeholder="Search author"
+            value={selectedKeys[0]}
+            onChange={(e) =>
+              setSelectedKeys(e.target.value ? [e.target.value] : [])
+            }
+            onPressEnter={() => confirm()}
+            style={{ width: 188, marginBottom: 8, display: "block" }}
+          />
+          <Button
+            type="primary"
+            onClick={() => confirm()}
+            size="small"
+            style={{ width: 90, marginRight: 8 }}
+          >
+            Search
+          </Button>
+          <Button
+            onClick={() => clearFilters()}
+            size="small"
+            style={{ width: 90 }}
+          >
+            Reset
+          </Button>
+        </div>
+      ),
+    },
+  ];
+
+  const handleSubmit = async ({
+    title,
+    description,
+    published_date,
+    author_id,
+  }: {
+    title: string;
+    description: string;
+    published_date: string;
+    author_id?: string;
+  }) => {
+    try {
+      if (selectedBook) {
+        const variables: any = {
+          id: selectedBook.id,
+          title: title,
+          description: description,
+          publishedDate: published_date,
+        };
+
+        if (author_id) {
+          variables.authorId = author_id;
```

Comment:
This seems confusing, `authorId` and `author_id`

Reply:
You're correct name should be consistent